api/src/services/assets.ts

```diff
@@ -66,9 +66,48 @@ export class AssetsService {
 		if (!exists) throw new ForbiddenException();
 
 		if (range) {
-			if (range.start >= file.filesize || (range.end && range.end >= file.filesize)) {
+			if (!range.start && !range.end) {
 				throw new RangeNotSatisfiableException(range);
 			}
+
+			if (range.end) {
+				if (range.end <= 0) {
+					throw new RangeNotSatisfiableException(range);
+				}
+
+				if (!range.start) {
+					// fetch chunk from tail
+					range.start = file.filesize - range.end;
+					range.end = file.filesize - 1;
+				}
+
+				if (range.end >= file.filesize) {
+					// fetch entire file
+					range.end = undefined;
```

Comment:
```suggestion
					range.end = file.filesize -1;
```

Reply:
all `// fetch entire file` should set `range.end = undefined` or `range.end = file.filesize - 1` consistently

Reply:
Implemented 👍